samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +515,109 @@ def write_response(self, result, params, messages):
       'result': result,
       'params': params
     }))
+    
+  def report_error(self, error):
+    self.write_response(error, {}, [])
 
   def write_room_parameters(self, room_id, client_id, messages, is_initiator):
     params = get_room_parameters(self.request, room_id, client_id, is_initiator)
     self.write_response('SUCCESS', params, messages)
+    
+  def handle_call(self, msg, room_id):
+    """Handles a new call request."""
+    if not util.has_msg_fields(msg, ((PARAM_CALLER_GCM_ID, basestring),
```

Comment:
Would it be cleaner to move all the requests into their own classes with parse methods? That way we can clean up the file(s) a bit since there won't be PARAM_* globals and these type checks. Instead we can have something like CallRequest.parse(msg).